pkg/scheduler/framework/plugins/serviceaffinity/service_affinity.go

```diff
@@ -293,16 +285,105 @@ func (pl *ServiceAffinity) Score(ctx context.Context, state *framework.CycleStat
+	if node == nil {
+		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("node not found"))
+	}
+
+	// Pods matched namespace,selector on current node.
+	var selector labels.Selector
+	if services, err := schedulerlisters.GetPodServices(pl.serviceLister, pod); err == nil && len(services) > 0 {
+		selector = labels.SelectorFromSet(services[0].Spec.Selector)
+	} else {
+		selector = labels.NewSelector()
+	}
+
+	if nodeInfo.Pods() == nil || len(nodeInfo.Pods()) == 0 || selector.Empty() {
+		return 0, nil
+	}
+	var score int64
+	for _, existingPod := range nodeInfo.Pods() {
+		// Ignore pods being deleted for spreading purposes
+		// Similar to how it is done for SelectorSpreadPriority
+		if pod.Namespace == existingPod.Namespace && existingPod.DeletionTimestamp == nil {
+			if selector.Matches(labels.Set(existingPod.Labels)) {
+				score++
+			}
+		}
+	}
+
+	return score, nil
 }
 
 // NormalizeScore invoked after scoring all nodes.
 func (pl *ServiceAffinity) NormalizeScore(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
-	// Note that priorityReduceFunction doesn't use priority metadata, hence passing nil here.
-	err := pl.priorityReduceFunction(pod, nil, pl.sharedLister, scores)
-	return migration.ErrorToFrameworkStatus(err)
+	reduceResult := make([]float64, len(scores))
+	for _, label := range pl.args.AntiAffinityLabelsPreference {
+		if err := pl.updateNodeScoresForLabel(pl.sharedLister, scores, reduceResult, label); err != nil {
+			return framework.NewStatus(framework.Error, err.Error())
+		}
+	}
+
+	// Update the result after all labels have been evaluated.
+	for i, nodeScore := range reduceResult {
+		scores[i].Score = int64(nodeScore)
+	}
+	return nil
+}
+
+// updateNodeScoresForLabel updates the node scores for a single label. Note it does not update the
+// original result from the map phase directly, but instead updates the reduceResult, which is used
+// to update the original result finally. This makes sure that each call to updateNodeScoresForLabel
+// receives the same mapResult to work with.
+// Why are doing this? This is a workaround for the migration from priorities to score plugins.
+// Historically the priority is designed to handle only one label, and multiple priorities are configured
+// to work with multiple labels. Using multiple plugins is not allowed in the new framework. Therefore
+// we need to modify the old priority to be able to handle multiple labels so that it can be mapped
+// to a single plugin.
+// TODO: This will be deprecated soon.
+func (pl *ServiceAffinity) updateNodeScoresForLabel(sharedLister schedulerlisters.SharedLister, mapResult framework.NodeScoreList, reduceResult []float64, label string) error {
+	var numServicePods int64
+	var labelValue string
+	podCounts := map[string]int64{}
+	labelNodesStatus := map[string]string{}
+	maxPriorityFloat64 := float64(framework.MaxNodeScore)
+
+	for _, hostPriority := range mapResult {
```

Comment:
use ```node``` instead of ```host``` here and all places below

Reply:
done